cmd/frontend/backend/external_services.go

```diff
@@ -41,6 +61,132 @@ func SyncExternalService(ctx context.Context, logger log.Logger, svc *types.Exte
 		}
 	}()
 
-	_, err = client.SyncExternalService(ctx, svc.ID)
+	_, err = e.repoupdaterClient.SyncExternalService(ctx, svc.ID)
 	return err
 }
+
+// ExcludeRepoFromExternalService excludes given repo from given external service config.
+//
+// Function is pretty beefy, what it does is:
+// - finds an external service by ID and checks if it supports repo exclusion
+// - adds repo to `exclude` config parameter and updates an external service
+// - triggers external service sync
+func (e *externalServices) ExcludeRepoFromExternalService(ctx context.Context, externalServiceID int64, repoID api.RepoID) (err error) {
+	tx, err := e.db.Transact(ctx)
+	if err != nil {
+		return err
+	}
+	defer func() { err = tx.Done(err) }()
+
+	externalServices := tx.ExternalServices()
+	externalService, err := externalServices.GetByID(ctx, externalServiceID)
+	if err != nil {
+		return err
+	}
+
+	logger := e.logger.Scoped("ExcludeRepoFromExternalService", "excluding a repo from external service config").With(
+		log.Int64("externalServiceID", externalServiceID),
+		log.Int32("repoID", int32(repoID)),
+	)
+
+	// If external service doesn't support repo exclusion, then return.
+	if !externalService.SupportsRepoExclusion() {
+		logger.Warn("Tried to exclude repo from external service, but its config does not support repo exclusion.")
```

Comment:
The other information is already part of the logger after the `logger.Scoped()` call.

```suggestion
		logger.Warn("external service does not support repo exclusion")
```